src/SDKs/Azure.Core/data-plane/Azure.Core/Pipeline/Policies/TaskExtensions.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Diagnostics;
+using System.Threading.Tasks;
+
+namespace Azure.Core.Pipeline.Policies
+{
+    internal static class TaskExtensions
+    {
+        public static T AssertCompleted<T>(this Task<T> task)
+        {
+            AssertTaskCompleted(task);
+            return task.GetAwaiter().GetResult();
+        }
+
+        public static void AssertCompleted(this Task task)
+        {
+            AssertTaskCompleted(task);
+            task.GetAwaiter().GetResult();
+        }
+
+        [Conditional("DEBUG")]
+        private static void AssertTaskCompleted(Task task)
```

Comment:
Should we consider removing this method?  I'm not sure that it adds clarity for the reader and replaces a straight-forward single line statement.   According to the [docs](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.debug.assert?view=netframework-4.7.2), the `Debug.Assert` has the conditional applied to it already.